tests/test_inference.py

```diff
@@ -1273,6 +1273,17 @@ def verify(self, sample_data, mismatch_rate=None, recombination_rate=None):
                 )
                 self.assertEqual(ancestor_data.provenances_timestamp[j], p.timestamp)
 
+            # Ancestors indicated in node metadata should have the same age as their node
+            ancestors_time = ancestor_data.ancestors_time[:]
+            num_ancestor_nodes = 0
+            for n in ancestors_ts.nodes():
+                if n.metadata:
+                    md = json.loads(n.metadata)
+                    if "ancestor" in md:
```

Comment:
We should be testing whether this is present for non-pc ancestors using the flags, right? Currently this would pass if no metadata was added at all.

Reply:
I'm not sure what you mean here, sorry. If no metadata was added at all, then at the end, num_ancestor_nodes would be 0, so the test would fail. I could assert that the nodes with metadata containing 'ancestor' are *not* PC nodes? But there isn't a flag that marks whether an node in the ancestors ts corresponds to an ancestor in the file. That's the point of the metadata, I guess.

Reply:
Well, every ancestor is either a PC ancestor OR an ancestor from the file, right? So, if the PC flag is not set, then metadata must contain this "ancestor" key. If it is, it must not contain this ancestor key.

Reply:
Right, so I've just pushed a change to check that the PC flag is not set on these ancestors. I could also check that PC nodes do *not* have that metadata field, I guess? Would you like that additional check? I think the assert at the end checks that we have caught all the ancestor nodes.

Reply:
Yes - what we want is:
```
if is_pc_ancestor(node):
     assert "ancestor" not in md
else:
      assert "ancestor" in md
```
Although, looking at it now, "ancestor" is perhaps not the best metadata key. What about ``ancestor_data_id`` - it's clear then that this is it's ID in the ancestor data file.

Reply:
Yes, I chose 'ancestor' because we use 'sample' as a metadata key in `get_augmented_ancestors_tree_sequence`, so I thought that should match. Perhaps we should change both of them to have `_id` on the end?

Reply:
> Yes - what we want is:
> 
> ```
> if is_pc_ancestor(node):
>      assert "ancestor" not in md
> else:
>       assert "ancestor" in md
> ```

I was trying to avoid being exhaustive, in case other nodes are will be added in the future (e.g. there is some mention of SRB nodes). I really just wanted to check that the nodes with 'ancestor' were valid, rather than validate at all the nodes.

Reply:
We should validate all nodes. If we change things in the future the tests *should* break - that's part of the point.

Reply:
Hmm, OK. Do you want both "ancestors" and "samples" metadata keys changed too?

Reply:
Yes, might be good to be explicit and change these to ``ancestor_data_id`` and ``sample_data_id`` to be clear that these are the IDs within the given files.

Reply:
If we are keeping `sample_data` and `ancestor_data`, you might want to close https://github.com/tskit-dev/tsinfer/issues/103, @jeromekelleher

Reply:
True, done.